fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/centralstore/CentralStore.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * CentralStore.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2020 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb.centralstore;
+
+import com.apple.foundationdb.record.RecordMetaData;
+import com.apple.foundationdb.record.RecordMetaDataBuilder;
+import com.apple.foundationdb.record.RecordMetaDataProto;
+import com.apple.foundationdb.record.metadata.Key;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordContext;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordStore;
+import com.apple.foundationdb.record.provider.foundationdb.MetaDataProtoEditor;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.function.Function;
+
+/**
+ * TODO: CentralStore
+ */
+public class CentralStore {
+    @Nonnull
+    private static final CentralStore INSTANCE = new CentralStore();
```

Comment:
Yes, definitely, we've gotten burned by this mistake in the past (we == me, with the reverse directory layer being static a singleton).  It may make the most sense to have a `FDBDatabase` own it's instance of the central store, however Scott D makes a good argument that we may even just want to let the client manage its own central store(s).